packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/RealmMapInternal.kt

```diff
@@ -657,6 +677,11 @@ internal class ManagedRealmDictionary<V> constructor(
                 }
         }
     }
+
+    override fun toString(): String = entries.joinToString { (key, value) -> "[$key,$value]" }
```

Comment:
Hmm, I can see where this is coming from, but it looks pretty dangerous, i.e. lets say you have dictionaries with 100's or 1000's of elements. I would probably avoid doing this, but there doesn't seem to be a great alternative either.

Since `toString` is generally used for debugging I would probably rather turn this into something like `RealmDictionary{owner=<classType>,objKey=<objectId>,version=<transactionVersion>}`. It would be easy to compute and you would still get some context for debugging.

Reply:
Since there was no `toString` implementation for the other collections I took the approach we use for Java lists but I agree it can easily backfire with large datasets. I think your proposal is fair, I would only add `size` for clarity too.